doc/common_pitfalls.rst

```diff
@@ -225,3 +225,346 @@ Below are some tips on avoiding data leakage:
   leakage as it ensures that the appropriate method is performed on the
   correct data subset. The pipeline is ideal for use in cross-validation
   and hyper-parameter tuning functions.
+
+.. _randomness:
+
+Controlling randomness
+======================
+
+Some scikit-learn objects are inherently random. These are usually estimators
+(e.g. :class:`~sklearn.ensemble.RandomForestClassifier`) and cross-validation
+splitters (e.g. :class:`~sklearn.model_selection.KFold`). The randomness of
+these objects is controlled via their `random_state` parameter, as described
+in the :term:`Glossary <random_state>`. This section expands on the glossary
+entry, and describes good practices and common pitfalls w.r.t. to this
+subtle parameter.
+
+.. note:: Recommendation summary
+
+    For an optimal robustness of cross-validation (CV) results, pass
+    `RandomState` instances when creating estimators, or leave `random_state`
+    to None. Passing integers to CV splitters is usually the safest option,
+    although `RandomState` instances are fine if you know what you are doing.
```

Comment:
Can we remove "if you know what you are doing"

Reply:
May I ask why?
I think it's important to stress that this is usually not a standard use case and that it can cause unexpected results.
Could you suggest an alternative?

Reply:
I think the words "if you know what you are doing"....  is not an objective statement of fact.

How about, "if you are familiar with xxx...  then you can use"  
It's much different than "knowing or not doing what you are doing"

>Passing integers to CV splitters is usually the safest option, although `RandomState` are fine...
 if one is familiar with the deeper concepts/structures/code and its implications...

Reply:
So many people in this field struggle with imposter syndrome.  When reading that, we don't want folks to ask themselves this question "Am I someone who knows what I am doing here?" or "Am I someone who doesn't know what I am doing here?"

We want people to know the specifics of what they can be familiar with to either do it this way or that way.

Reply:
Thanks for the reply, I'll remove and use

> passing `RandomState` instances to splitters may
    sometimes be useful to achieve very specific use-cases

(For what it's worth, I think that "I've no idea what I'm doing" is something people should feel comfortable saying. It's very normal IMHO - and it's actually one of my go-to commit messages ^^ )